pipeline.py

```diff
@@ -0,0 +1,60 @@
+# type: ignore
+# pylint: disable=no-value-for-parameter,import-outside-toplevel,import-error,no-member
+from typing import List, Literal, Optional
+import click
+from kfp import dsl, compiler
+from kfp.kubernetes import use_config_map_as_env, use_secret_as_env
+
+K8S_NAME = "kfp-model-server"
+MOCKED_STAGES = ['sdg', 'train', 'eval']
+
+def pipeline_wrapper(mock: List[Optional[Literal[MOCKED_STAGES]]]):
+    """Wrapper for KFP pipeline, which allows for mocking individual stages."""
+    if 'sdg' in mock:
+        from sdg.faked import git_clone_op, sdg_op
+    else:
+        from sdg import git_clone_op, sdg_op
+
+
+    @dsl.pipeline(
+        display_name="InstructLab",
+        name="instructlab",
+        description="InstructLab pipeline",
+    )
+    def pipeline(
+        num_instructions_to_generate: int = 2,
+        repo_url: str = "https://github.com/instructlab/taxonomy.git",
+        repo_branch: Optional[str] = None,
+        repo_pr: Optional[int] = None,
+    ):
+        git_clone_task = git_clone_op(
+            repo_branch=repo_branch, repo_pr=repo_pr, repo_url=repo_url
+        )
+
+        sdg_task = sdg_op(
```

Comment:
Should we consider throwing an error when `repo_pr` or `repo_branch` is None? Because that will lead to the `no leaf node` error in the sdg.

Reply:
If both are none, it should run against `"empty"`. That should not throw that error. Weird.. anyways, lets log that and fix that as a separate PR, can you please file an issue for me?